packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -19,17 +20,33 @@ interface ComboboxContentProps {
    * actions
    */
   readonly children: ReactElement | ReactElement[];
+
+  /**
+   * pre selected option
+   * @default ""
+   * @optional
+   * @type string
+   * */
+  readonly preSelectedOption?: string;
 }
 
 export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
+  const { preSelectedOption } = props;
   const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
+  const [selectedOption, setSelectedOption] = useState<string>(
+    preSelectedOption || "",
+  );
   const { open } = React.useContext(ComboboxContext);
 
   const filteredOptions = props.options.filter(option =>
     option.toLowerCase().includes(searchValue.toLowerCase()),
   );
+  let invalidPreSelectedOption = false;
+  if (preSelectedOption && !props.options.includes(preSelectedOption)) {
+    invalidPreSelectedOption = true;
+  }
 
+  useAssert(invalidPreSelectedOption, "Preselected option must be in options");
```

Comment:
wondering how heavy handed we need to be here. if someone puts `"nonsense.jpg"` as the `preSelectedOption` what happens?

the component still renders happily, they can still use it as normal once they open it and choose something that **is** in the list

the only thing that doesn't work of course is that nothing is preselected. is that punishment enough or enough incentive to give it something it can find & use?

Reply:
hm ya I considered this definitely open to opinions on how strict to be here. I figured we would want a pre selected option that doesn't exist to to throw an error but open to your opinion 🤔